service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -94,10 +96,12 @@ public MetadbSample fetchAndLoadSampleDetails(MetadbSample sample) throws Except
     @Override
     public MetadbSample getMetadbSample(UUID metadbSampleId) throws Exception {
         MetadbSample sample = sampleRepository.findSampleById(metadbSampleId);
+        if (sample == null) {
+            return null;
+        }
         sample.setSampleMetadataList(sampleRepository.findSampleMetadataListBySampleId(metadbSampleId));
         for (SampleMetadata s: sample.getSampleMetadataList()) {
             s.setMetaDbPatientId(patientService.getPatientIdBySample(metadbSampleId));
```

Comment:
so after doing some of my own testing as well i confirmed that you don't need the patient uuid at the sample **metadata** level so we can remove this for-block (and others like this throughout the sample service) and instead just fetching and setting the patient of the `MetadbSample`

```
MetadbPatient patient = patientService.getPatientByCmoPatientId(cmoPatientId);
sample.setPatient(patient);
```